Plutarch/Internal.hs

```diff
@@ -747,6 +751,32 @@ compile :: Config -> ClosedTerm a -> Either Text Script
 compile config t = case asClosedRawTerm t of
   TermMonad (ReaderT t') -> Script . UPLC.Program () uplcVersion . compile' <$> t' config
 
+{- | As 'compile', but performs UPLC optimizations. Furthermore, this will
+always elide tracing (as if with 'NoTracing').
+
+@since WIP
+-}
+compileOptimized ::
+  forall (a :: S -> Type).
+  (forall (s :: S). Term s a) ->
```

Comment:
I know it's a GHC misfeature, but `ClosedTerm a` would be better for consistency